sale_require_purchase_order_number/models/stock_picking.py

```diff
@@ -9,42 +9,42 @@
 class StockPicking(models.Model):
     _inherit = "stock.picking"
 
-    @api.one
     @api.depends('group_id', 'sale_id')
-    def _get_purchase_order_number(self):
-        if self.manual_purchase_order_number:
-            self.purchase_order_number = self.manual_purchase_order_number
-        else:
-            self.purchase_order_number = self.sale_id.purchase_order_number
+    def _compute_purchase_order_number(self):
+        for rec in self:
+            rec.purchase_order_number = rec.manual_purchase_order_number\
+                if rec.manual_purchase_order_number\
+                else rec.sale_id.purchase_order_number
 
-    @api.one
-    def _set_purchase_order_number(self):
-        self.manual_purchase_order_number = self.purchase_order_number
+    @api.multi
+    def _inverse_purchase_order_number(self):
+        for rec in self:
+            rec.manual_purchase_order_number = rec.purchase_order_number
 
     require_purchase_order_number = fields.Boolean(
         string='Sale Require Origin',
         related='partner_id.require_purchase_order_number',
-        readonly=True,)
+        readonly=True,
+    )
     manual_purchase_order_number = fields.Char(
         'Purchase Order Number',
         states={'cancel': [('readonly', True)],
-                'done': [('readonly', True)]})
+                'done': [('readonly', True)]},
+    )
     purchase_order_number = fields.Char(
-        compute='_get_purchase_order_number',
-        inverse='_set_purchase_order_number',
-        string='Purchase Order Number',
+        compute='_compute_purchase_order_number',
+        inverse='_inverse_purchase_order_number',
         states={'cancel': [('readonly', True)],
-                'done': [('readonly', True)]})
-    code = fields.Selection(
-        related='picking_type_id.code',
-        string='Operation Type',
-        readonly=True)
+                'done': [('readonly', True)]},
+    )
 
     @api.multi
-    def do_new_transfer(self):
-        if self.require_purchase_order_number and self.code == 'outgoing':
-            if not self.purchase_order_number:
+    def action_done(self):
+        for rec in self.filtered(
```

Comment:
same her enot need to make the loop. only save the filtered values in a variable and then ask if exist raise error